### ISSUE TITLE
4.0b4 Crash with GeoRAM 4096Kb

### ISSUE THREAD

**Issue by bluecursor** [dirkwhoffmann/virtualc64#631]
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
Performing @selector(attachGeoRamAction:) from sender NSMenuItem 0x60000029e290
Assertion failed: (false), function attachGeoRamCartridge, file /Users/hoff/virtualc64/Emulator/Peripherals/ExpansionPort.cpp, line 241.

**Comment by bluecursor** [comment]
Also would be nice to have a check mark on which size is being used...

**Comment by dirkwhoffmann** [comment]
The crash was a pretty stupid bug in `attachGeoRamCartridge`:

```C++
void
ExpansionPort::attachGeoRamCartridge(usize kb)
{
    debug(EXP_DEBUG, "Attaching GeoRAM cartridge (%zu KB)", kb);

    if (kb != 64 && kb != 256 && kb != 512 && kb != 1024 && kb != 2048) {
        assert(false);
    }
    ...
```

I've replaced this by

```C++
void
ExpansionPort::attachGeoRamCartridge(usize kb)
{
    debug(EXP_DEBUG, "Attaching GeoRAM cartridge (%zu KB)", kb);

     // kb must be a power of two between 64 and 4096
    if (kb < 64 || kb > 4096 || (kb & kb - 1)) assert(false);
    ...
```

I agree there should be a checkmark on the RAM size. I'll add one.

Snapshots follow Apple's Time Machine principle: "We save everything". Hence, yes, the GeoRam contents is supposed to be part of snapshots, even if no battery is emulated. The battery comes into play when the machine is reset. Battery backed up GeoRam will keep the data. A GeoRam without a (virtual) battery will loose it.

**Comment by dirkwhoffmann** [comment]
<img width="316" alt="Bildschirmfoto 2021-02-02 um 16 13 07" src="https://user-images.githubusercontent.com/12561945/106620521-f02b2000-6571-11eb-82fc-9725d7ed1970.png">

Now, a checkmark is shown.

**Comment by dirkwhoffmann** [comment]
Should be fixed in v4.0b5.
